tools/topology/topology2/platform/intel/speaker-echo-ref.conf

```diff
@@ -0,0 +1,66 @@
+Define {
+	ECHO_REF_STREAM_NAME	"Echo reference"
+}
+
+Object.Dai {
+	# override amp instance's i2s config for echo reference
+	SSP."1" {
```

Comment:
@yongzhi1 lets not do this. My alsauilts PR will not allow object overrides. You need to set the direction and lbm mode to duplex is INCLUDE_ECHO_REF is true in the top-level file.

Reply:
The lbm_mode is the only valid value for SSP quirks, I will think about your suggestion.
`
			!valid_values [
				"lbm_mode"
			]
`

Reply:
@yongzhi1 what you need is something like this. You will need the latest upstream alsa-utils for this to work.
```
diff --git a/tools/topology/topology2/cavs-rt5682.conf b/tools/topology/topology2/cavs-rt5682.conf
index b04957d99..e13bc362c 100644
--- a/tools/topology/topology2/cavs-rt5682.conf
+++ b/tools/topology/topology2/cavs-rt5682.conf
@@ -116,7 +116,15 @@ Object.Dai {
        SSP."1" {
                id              $SPK_ID
                dai_index       $SPEAKER_SSP_DAI_INDEX
+IncludeByKey.INCLUDE_ECHO_REF {
+"false" {
                direction       "playback"
+       }
+"true" {
+               direction       "duplex"
+               quirks          "lbm_mode"
+       }
+}
                name            $SPEAKER_CODEC_NAME
                default_hw_conf_id      0
                sample_bits             32
```

Reply:
Ack, fortunately I waited a while before attempting the changes, the new capability arrived to rescue at the right moment: 
https://github.com/alsa-project/alsa-utils/commit/1350900246e0462b584331be0df24e177e7b25be